engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +470,68 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  private static void makeMDCProperties()
+  {
+    if (setMDCProperties) {
+
+      MDC.put("node", getCurrentHost());
+
+      String value = System.getenv("HADOOP_USER_NAME");
+      if (value != null) {
+        MDC.put("user", value);
+      }
+
+      value = System.getenv("CONTAINER_ID");
+      if (value != null) {
+        MDC.put("containerId", value);
+      }
+
+      Properties properties = System.getProperties();
+      String longName = LogicalPlan.APPLICATION_NAME.getLongName();
+      if (properties.contains(longName)) {
+        MDC.put("application", properties.getProperty(longName));
+      }
+
+      if (MDC.get("service") == null) {
```

Comment:
MDC properties are static in Java process. If both app master and containers are in the same JVM, both of them will have the same service name.